include/pybind11/attr.h

```diff
@@ -176,16 +176,19 @@ void keep_alive_impl(size_t Nurse, size_t Patient, function_call &call, handle r
 
 /// Internal data structure which holds metadata about a keyword argument
 struct argument_record {
-    const char *name;  ///< Argument name
-    const char *descr; ///< Human-readable version of the argument value
-    handle value;      ///< Associated Python object
+    const char *name;      ///< Argument name
+    const char *descr;     ///< Human-readable version of the argument value
+    handle value;          ///< Associated Python object
```

Comment:
Should this be made to a std::optional<handle> rather than adding a boolean? Using an optional would enforce that no value is hold when value_is_nullptr is True.

Reply:
I agree, but

* that would be a much more intrusive change,
* especially also because we still care (***) about C++11 compatibility. `std::optional` requires C++17.

(***) Because I'm still keeping pywrapcc very closely aligned with upstream pybind11 master.

Net change:

I added a comment to explain why `value_is_nullptr` exists (safety).